packages/node_modules/@node-red/nodes/core/common/20-inject.js

```diff
@@ -62,34 +76,27 @@ module.exports = function(RED) {
           node.repeaterSetup();
         }
 
-        this.on("input",function(msg) {
-            msg.topic = this.topic;
-            if (this.payloadType !== 'flow' && this.payloadType !== 'global') {
+        this.on("input", function(msg) {
+            var errors = [];
+
+            this.props.forEach(p => {
+                var property = p.p;
+                var value = p.v ? p.v : '';
+                var valueType = p.vt ? p.vt : 'str';
+
+                if (!property) return;
+
                 try {
-                    if ( (this.payloadType == null && this.payload === "") || this.payloadType === "date") {
-                        msg.payload = Date.now();
-                    } else if (this.payloadType == null) {
-                        msg.payload = this.payload;
-                    } else if (this.payloadType === 'none') {
-                        msg.payload = "";
-                    } else {
-                        msg.payload = RED.util.evaluateNodeProperty(this.payload,this.payloadType,this,msg);
-                    }
-                    this.send(msg);
-                    msg = null;
-                } catch(err) {
-                    this.error(err,msg);
+                    msg[property] = RED.util.evaluateNodeProperty(value, valueType, this, msg);
```

Comment:
It seems some of this logic for handling the payload was later built into `evaluateNodeProperty`. I simplified it by relying on `evaluateNodeProperty` to handle all of the TypedInput values correctly - and in testing it seems that it does.

Reply:
Aha - yes this to be based on `RED.util.setMessageProperty(msg,property,value)`  in order to handle the sub properties correctly.